### DISCUSSION TITLE
Headless login

### DISCUSSION THREAD

**Discussion by hhsecond** [supabase/supabase#15616] [Questions]
I am building an app that is using supabase backend. It has user authentication and a bunch of stuff my users would do on the web app. But it also has a CLI which would be executed in different environments including headless servers. Because my users would choose any type of authentication (social login including) The typical workflow I had in mind was, I generate an api key for the user from my server. User use that key in the headless environment to authenticate

```
mycli login --key <key-I-created>
```

My assumption was that, this requests go to my middleware, and then it will verify the api key I created for them, and then generate the access token and refresh token for the user and respond it to the cli. From that point onwards, CLI uses those tokens to communicate to the supabase server directly. It doesn't seem possible. What are my options?

**Comment by j4w8n** [comment]
I'm finishing up a proof-of-concept for doing something similar. I'll probably put it on github once I'm done.
